functions/v2/http-logging/main.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# [START functions_structured_logging]
+import google.cloud.logging
+from google.cloud.logging.handlers import StructuredLogHandler
+from google.cloud.logging_v2.handlers import setup_logging
+
+
+def structured_logging(request):
+    logging_client = google.cloud.logging.Client()
+
+    handler = StructuredLogHandler()
+    setup_logging(handler)
+
+    logger = logging_client.logger("New-Structured-Log")
+
+    msg = (f"{request.headers}"
+           f"{request.get_data().decode()}"
+           )
```

Comment:
Hmm, this shouldn't be needed. Let's check with the Python Observability expert

Reply:
The httpRequest data should be attached automatically, assuming there's a valid [flask](https://github.com/googleapis/python-logging/blob/28d48b99cad464d4cbcb1ddeb0dce9f2a5403084/google/cloud/logging_v2/handlers/_helpers.py#L63) or [django](https://github.com/googleapis/python-logging/blob/28d48b99cad464d4cbcb1ddeb0dce9f2a5403084/google/cloud/logging_v2/handlers/_helpers.py#L94) request in scope

You can build the http request object manually if you want, but this format seems incorrect. It should be a dictionary (or proto struct) containing [valid LogEntry HttpRequest fields](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest)

```
    http_request = {
        "requestMethod": request.method,
        "requestUrl": request.url,
        "userAgent": request.user_agent.string,
    }
```